### PR TITLE
chore: Fix Logging in torch_compile path 

### PR DESCRIPTION
**Pull request by peri044**
# Description

Set the log level in torch_compile path at the start of compilation

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/backend/backends.py [hunk 1]
```diff
@@ -20,6 +20,12 @@
 def torch_tensorrt_backend(
     gm: torch.fx.GraphModule, sample_inputs: Sequence[torch.Tensor], **kwargs: Any
 ) -> torch.nn.Module:
+    # Set log level at the top of compilation (torch_tensorrt.dynamo)
+    if "options" in kwargs:
+        if "debug" in kwargs["options"]:
+            if logger.parent:
+                logger.parent.setLevel(logging.DEBUG)
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/backend/backends.py:27]: Could this be moved into the `parse_dynamo_kwargs` function in `dynamo/utils`, under the `if settings.debug` section? If not, could the logic be changed to:
```python
if ((("options" in kwargs and "debug" in kwargs["options"] and kwargs["options"]["debug"])
   or ("debug" in kwargs and kwargs["debug"]))
   and logger.parent):
    logger.parent.setLevel(logging.DEBUG)
```

**Review by gs-olive — COMMENTED** [review]
Changes look good - just made a conditional logic suggestion and question.

File: py/torch_tensorrt/dynamo/backend/backends.py (hunk 1 above)
Comment (peri044) [reply py/torch_tensorrt/dynamo/backend/backends.py]: This is much cleaner. Thanks. I'd like to set the debug level on the entry points of compilation (for both export and compile), that way we don't have to set it anywhere else. Also, made the `CompilationSettings` object logging to info (similar to export).

**Review by gs-olive — APPROVED** [review]
Looks good to me!
